src/gmail.js

```diff
@@ -1329,6 +1334,8 @@ var Gmail = function(localJQuery) {
                     const fd_email_timestamp = Number.parseInt(fd_email["2"]["17"]);
                     const fd_email_date = new Date(fd_email_timestamp);
 
+                    const fd_email_is_draft = api.tools.parse_fd_bv_is_draft(fd_email2["4"]);
```

Comment:
In this part of the codebase, in this particular function, you have to keep in mind that we are actually intercepting/hijacking Gmail XHR requests, and any failure in our code at this point will cause entire Gmail to stop working. Full stop.

**As such, this code needs to be 100% bullet-proof against any sort of failure.**

We can kinda accept that Gmail.JS will stop working, but we should never accept that using Gmail.JS in an extension causes *Gmail itself* to stop working. At least, that has been the priority for this part of the code so far.

So, with that said: This new line seems to be one of few using the `fd_email2` variable, the other being `api.tools.parse_fd_bv_contact()`.

I'd be more comfortable with adding this new function into this loop, if it implements the same kind of safeguards as `api.tools.parse_fd_bv_contact()` does (that is, `try{ ... } catch(){ return false; }`) .

Reply:
Thanks - totally agree on the philosophy.  I did not think it was possible for this code to cause a failure - I think we're guaranteed `fd_email2["4"]` itself will not throw because (a) we already called `fd_email2["2"]` for `parse_fd_bv_contact` and (b) everything extends an object so getting a property is always safe. If it is not of type `Array`, then `Array.isArray` will return `false` and we'll exit.  If it is of type `Array` then `.includes` will be well defined. I mean, I suppose it's possible someone overrides `Array` to remove `include`, but that seemed fairly intentionally malicious at that point :)

Anyway, let me know if you think I've missed something here - also happy to just add it anyways if you feel strongly since its presumably irrelevant from a performance standpoint.

Reply:
In theory I agree with you. I’m practice, let’s add a try/catch just in case 😅